habitat/tasks/rearrange/envs/utils.py

```diff
@@ -0,0 +1,427 @@
+
+    def calc_ik(self, targ_ee):
+        """
+        targ_ee is in ROBOT COORDINATE FRAME NOT IN EE COORDINATE FRAME
+        """
+        js = p.calculateInverseKinematics(
+            self.robo_id, self.pb_link_idx, targ_ee, physicsClientId=self.pc_id
+        )
+        return js[: self._arm_len]
+
+
+@attr.s(auto_attribs=True, kw_only=True)
+class CollDetails:
+    obj_scene_colls: int = 0
+    robo_obj_colls: int = 0
+    robo_scene_colls: int = 0
+
+
+def rearrang_collision(
+    colls,
+    snapped_obj_id,
+    count_obj_colls,
+    verbose=False,
+    ignore_names=[],
+    ignore_base=True,
+):
+    """
+    Defines what counts as a collision for the Rearrange environment execution
+    """
+    # Filter out any collisions from the base
+    if ignore_base:
+        colls = [
+            x
+            for x in colls
+            if not ("base" in x[0]["link"] or "base" in x[1]["link"])
+        ]
+
+    def should_ignore(x):
+        for ignore_name in ignore_names:
+            if coll_name(x, ignore_name):
+                return True
+        return False
+
+    # Filter out any collisions with the ignore objects
+    colls = [x for x in colls if not should_ignore(x)]
+
+    # Check for robot collision
+    robo_obj_colls = 0
+    robo_scene_colls = 0
+    robo_scene_matches = get_collision_matches("fetch", colls, "name")
+    for match in robo_scene_matches:
+        urdf_on_urdf = (
+            match[0]["type"] == "URDF" and match[1]["type"] == "URDF"
+        )
+        with_stage = coll_prop(match, "Stage", "type")
+        fetch_on_fetch = (
+            match[0]["name"] == "fetch" and match[1]["name"] == "fetch"
+        )
+        if fetch_on_fetch:
+            continue
+        if urdf_on_urdf or with_stage:
+            robo_scene_colls += 1
+        else:
+            robo_obj_colls += 1
+
+    # Checking for holding object collision
+    obj_scene_colls = 0
+    if count_obj_colls:
+        if snapped_obj_id is not None:
+            matches = get_collision_matches(
+                "id %i" % snapped_obj_id, colls, "link"
+            )
+            for match in matches:
+                if coll_name(match, "fetch"):
+                    continue
+                obj_scene_colls += 1
+
+    total_colls = robo_obj_colls + robo_scene_colls + obj_scene_colls
+    return total_colls > 0, CollDetails(
+        obj_scene_colls=min(obj_scene_colls, 1),
+        robo_obj_colls=min(robo_obj_colls, 1),
+        robo_scene_colls=min(robo_scene_colls, 1),
+    )
+
+
+def get_nav_mesh_settings(agent_config):
+    return get_nav_mesh_settings_from_height(agent_config.HEIGHT)
+
+
+def get_nav_mesh_settings_from_height(height):
+    navmesh_settings = NavMeshSettings()
+    navmesh_settings.set_defaults()
+    navmesh_settings.agent_radius = 0.4
+    navmesh_settings.agent_height = height
+    navmesh_settings.agent_max_climb = 0.05
+    return navmesh_settings
+
+
+def convert_legacy_cfg(obj_list):
```

Comment:
Are these cfgs in question here ones that we control?  If so we should just ship the converted ones.